### PR TITLE
No unwrap in lib

### PR DESCRIPTION
**Pull request by matthiasbeyer**
**This is WIP**.

I'm working on this PR for no `unwrap()` calls in the `lib.rs` file, so I can use this as wrapper around the git2 crate as this seems to provide the convenience I need.

I'm to tired right now to finish this, but I guess I will finish it tomorrow or so.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -117,33 +117,34 @@ pub fn push(repo: &str, remote_name: &str, branches: &[String]) -> Result<(), Er
...
 
     for (branch, _) in branches {
-        let name = branch.name().expect("Invalid branch name");
-        let name = name.expect("Branch name not UTF-8");
+        let name = try!(branch.name());
+        let name = try!(name.ok_or(Error::from_str("Could not find branch name")));
 
         if name != short {
-            println!("  {}", name);
+            v.push(format!("  {}", name));
```
Comment (badboy) [inline src/lib.rs]: Only push the name, do the formatting/indenting in main.rs

**Comment by badboy** [comment]
Highly appreciated!

**Comment by badboy** [comment]
Please leave a comment if you consider this done

**Comment by matthiasbeyer** [comment]
Cleaned up history. Should be fine to merge now, I guess.

**Comment by badboy** [comment]
Hm, we might turn the thing returned from `branch` into an enum someday, to better separate between different branch types.
For now this looks good. We don't really have tests though.

@not-a-robot r+

**Comment by ghost** [comment]
:pushpin: Commit daa121e has been approved by `badboy`

**Comment by ghost** [comment]
:hourglass: Testing commit daa121e with merge 1693dd7...

**Comment by ghost** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/badboy/git2-commit-rs/builds/149149951)
Approved by: badboy
Pushing 1693dd7230e5e4a4af31bf3708557ed600daf071 to master...

**Pull request merged**